STYLEGUIDE.md

```diff
@@ -0,0 +1,133 @@
+
+# Typedefs styleguide
+
+This styleguide aims to answer questions relevants to the cosmetics related to 
+writing Idris programs.
+
+Since idris is a relatively young language, there is no general convention 
+about its style. This document should serve as reference for current and future 
+users of the typedefs codebase.
+
+## Line length: 120
+
+The maximum line length is 120 characters. In today's wide screen monitors 
+there is no reason to limit ourselves to 80 characters, yet, having a maximum
+limit helps avoiding degenerate situations.
+
+## Indent style: 2 spaces
+
+Idris makes uses of some whitespace sensitive construction (typically the `do`
+notation), and also inherits from a culture of ascii-art code from Agda and
+Haskell. In order to allow this level of visual customisation and ensure
+consistency across editors, spaces are used.
+
+The language does not benefit from more than 2 space of indentation in most
+cases. Thought special constructions lend themselve to more spaces for 
+indentation.
+
+### Indentation: where clauses
+
+``` idris
+fn : String
+fn = g
+  where g : String
+        g = "hello"
+```
+
```

Comment:
Currently, one of the styles I see in the codebase (e.g. [here](https://github.com/typedefs/typedefs/blob/0050cdcf1c19ce9bfc53ac35dcad6512be611e6b/src/Parse.idr)) is this:
```idris
fn : String
fn = g
  where
  g : String
  g = "hello"
```
Marco and myself had a discussion on how to do this in the AQL/CQL Haskell codebase, also considering the fact that ~~PureScript requires an indent after where~~. We settled on a 2-space indent after `where`, like this:
```idris
fn : String
fn = g
  where
    g : String
    g = "hello"
```
It may make sense to do the same for Typedefs in order to keep things memorable. (I guess we'd have to indent other constructs similarly.) This has the added benefit of consuming less horizontal space. I am aware that Idris codebases are slightly different beasts, so there may be issues with this, not sure.

Reply:
There is no issue with this, it makes just as much sense. I'll change the description accordingly

Reply:
I agree with 2 spaces indentation after `where`